Marlin/Marlin_main.cpp

```diff
@@ -848,9 +848,9 @@ static bool drain_injected_commands_P() {
  * Aborts the current queue, if any.
  * Note: drain_injected_commands_P() must be called repeatedly to drain the commands afterwards
  */
-void enqueue_and_echo_commands_P(const char * const pgcode) {
+bool enqueue_and_echo_commands_P(const char * const pgcode) {
   injected_commands_P = pgcode;
-  drain_injected_commands_P(); // first command executed asap (when possible)
+  return drain_injected_commands_P(); // first command executed asap (when possible)
```

Comment:
The status returned by `drain_injected_commands_P` only indicates that there are more commands to be queued. It does not indicate success or failure.